packages/util/src/account.ts

```diff
@@ -244,14 +239,25 @@ export const isValidPublic = function (publicKey: Buffer, sanitize: boolean = fa
   assertIsBuffer(publicKey)
   if (publicKey.length === 64) {
     // Convert to SEC1 for secp256k1
-    return publicKeyVerify(Buffer.concat([Buffer.from([4]), publicKey]))
+    // Automatically checks whether point is on curve
+    try {
+      Point.fromHex(Buffer.concat([Buffer.from([4]), publicKey]))
+      return true
+    } catch (e) {
+      return false
+    }
   }
 
   if (!sanitize) {
     return false
   }
 
-  return publicKeyVerify(publicKey)
+  try {
+    Point.fromHex(publicKey)
+    return true
+  } catch (e) {
+    return false
```

Comment:
this also seems to be a last uncovered line, if it's possible to add an invalid non-64 byte public key length, then diff coverage should be 100% :)